packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -18,25 +23,90 @@ const sfdxCoreExports = vscode.extensions.getExtension(
   'salesforce.salesforcedx-vscode-core'
 )!.exports;
 const {
+  channelService,
+  notificationService,
   telemetryService,
   SfdxCommandlet,
   EmptyParametersGatherer,
-  SfdxWorkspaceChecker
+  SfdxWorkspaceChecker,
+  sfdxCoreSettings
 } = sfdxCoreExports;
 
+enum PreviewPlatformType {
+  Desktop = 1,
+  Android,
+  iOS
+}
+
+const enum PlatformName {
+  desktop = 'Desktop',
+  android = 'Android',
+  ios = 'iOS'
+}
+
+interface PreviewQuickPickItem extends vscode.QuickPickItem {
+  label: string;
+  detail: string;
+  alwaysShow: boolean;
+  picked: boolean;
+  id: PreviewPlatformType;
+  defaultTargetName: string;
+  platformName: string;
+}
+
+export const platformOptions: PreviewQuickPickItem[] = [
+  {
+    label: nls.localize('force_lightning_lwc_preview_desktop_label'),
+    detail: nls.localize('force_lightning_lwc_preview_desktop_description'),
+    alwaysShow: true,
+    picked: true,
+    id: PreviewPlatformType.Desktop,
+    platformName: PlatformName.desktop,
+    defaultTargetName: ''
+  },
+  {
+    label: nls.localize('force_lightning_lwc_android_label'),
+    detail: nls.localize('force_lightning_lwc_android_description'),
+    alwaysShow: true,
+    picked: false,
+    id: PreviewPlatformType.Android,
+    platformName: PlatformName.android,
+    defaultTargetName: 'SFDXEmulator'
+  },
+  {
+    label: nls.localize('force_lightning_lwc_ios_label'),
+    detail: nls.localize('force_lightning_lwc_ios_description'),
+    alwaysShow: true,
+    picked: false,
+    id: PreviewPlatformType.iOS,
+    platformName: PlatformName.ios,
+    defaultTargetName: 'SFDXSimulator'
+  }
+];
+
 const logName = 'force_lightning_lwc_preview';
 const commandName = nls.localize('force_lightning_lwc_preview_text');
+const sfdxMobilePreviewCommand = 'force:lightning:lwc:preview';
+const rememberDeviceKey = 'rememberDevice';
```

Comment:
Consider moving the keys (rememberDeviceKey, ...), configuration defaults and functions like getRememberedDevice / updateRememberedDevice / isMobileEnabled into a different class, for example `service/previewService.ts`.

Reasoning: workspaceUtils doesn't provide much abstraction there. We are still seeing a lot of related names taking up the "global" (actually module, but this file is becoming bigger) namespace here. And `workspace.getConfiguration('salesforcedx-vscode-lwc').get('rememberDevice')` wouldn't be much less readable.

I think the right level of abstraction would be on the preview related configuration.